lib/ckeditor/hooks/action_policy.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'action_policy'
+
+module Ckeditor
+  module Hooks
+    # This adapter is for the Pundit[https://github.com/elabs/pundit] authorization library.
+    # You can create another adapter for different authorization behavior, just be certain it
+    # responds to each of the public methods here.
+    class ActionPolicyAuthorization
+      include Ckeditor::Helpers::Controllers
+
+      # See the +authorize_with+ config method for where the initialization happens.
+      def initialize(controller)
+        @controller = controller
+        # use ckeditor_current_user instead of default current_user so it works with
+        # whatever current user method is defined with Ckeditor
+        @controller.class.authorize :user, through: :ckeditor_current_user
+      end
+
+      # This method is called in every controller action and should raise an exception
+      # when the authorization fails. The first argument is the name of the controller
+      # action as a symbol (:create, :destroy, etc.). The second argument is the actual model
+      # instance if it is available.
+      def authorize(action, model_object = nil)
+        @controller.authorize!(model_object)
```

Comment:
Should there be an action?

```suggestion
        @controller.authorize!(model_object, to: :"#{action}?")
```

Reply:
I don’t think it’s necessary. We can get the name of the action [dynamically](https://github.com/palkan/action_policy/blob/aa686b635e8705992084b43975a0738d41fd8479/lib/action_policy/rails/controller.rb#L46)